x/bonds/internal/types/msgs.go

```diff
@@ -73,31 +74,31 @@ func NewMsgCreateBond(token, name, description string, creatorDid did.Did,
 	}
 }
 
-func (msg MsgCreateBond) ValidateBasic() sdk.Error {
+func (msg MsgCreateBond) ValidateBasic() error {
 	// Check if empty
 	if strings.TrimSpace(msg.BondDid) == "" {
-		return ErrArgumentCannotBeEmpty(DefaultCodespace, "BondDid")
+		return sdkerrors.Wrap(ErrArgumentCannotBeEmpty, "bond DID")
 	} else if strings.TrimSpace(msg.Token) == "" {
-		return ErrArgumentCannotBeEmpty(DefaultCodespace, "Token")
+		return sdkerrors.Wrap(ErrArgumentCannotBeEmpty, "token")
 	} else if strings.TrimSpace(msg.Name) == "" {
-		return ErrArgumentCannotBeEmpty(DefaultCodespace, "Name")
+		return sdkerrors.Wrap(ErrArgumentCannotBeEmpty, "name")
 	} else if strings.TrimSpace(msg.Description) == "" {
-		return ErrArgumentCannotBeEmpty(DefaultCodespace, "Description")
+		return sdkerrors.Wrap(ErrArgumentCannotBeEmpty, "description")
 	} else if strings.TrimSpace(msg.CreatorDid) == "" {
-		return ErrArgumentCannotBeEmpty(DefaultCodespace, "CreatorDid")
+		return sdkerrors.Wrap(ErrArgumentCannotBeEmpty, "creator DID")
 	} else if len(msg.ReserveTokens) == 0 {
-		return ErrArgumentCannotBeEmpty(DefaultCodespace, "Reserve token")
+		return sdkerrors.Wrap(ErrArgumentCannotBeEmpty, "reserve tokens")
 	} else if msg.FeeAddress.Empty() {
-		return ErrArgumentCannotBeEmpty(DefaultCodespace, "Fee address")
+		return sdkerrors.Wrap(ErrArgumentCannotBeEmpty, "fee address")
 	} else if strings.TrimSpace(msg.FunctionType) == "" {
-		return ErrArgumentCannotBeEmpty(DefaultCodespace, "Function type")
+		return sdkerrors.Wrap(ErrArgumentCannotBeEmpty, "function type")
 	}
 	// Note: FunctionParameters can be empty
 
 	// Check that bond token is a valid token name
 	err := CheckCoinDenom(msg.Token)
 	if err != nil {
-		return ErrInvalidCoinDenomination(DefaultCodespace, msg.Token)
+		return err
```

Comment:
`ErrInvalidCoinDenomination`?

Reply:
We can just use the error that CheckCoinDenom is returning